xarray/tests/test_distributed.py

```diff
@@ -34,29 +40,96 @@
 if has_h5netcdf:
     ENGINES.append('h5netcdf')
 
+NC_FORMATS = {'netcdf4': ['NETCDF3_CLASSIC', 'NETCDF3_64BIT_OFFSET',
+                          'NETCDF3_64BIT_DATA', 'NETCDF4_CLASSIC', 'NETCDF4'],
+              'scipy': ['NETCDF3_CLASSIC', 'NETCDF3_64BIT'],
+              'h5netcdf': ['NETCDF4']}
+TEST_FORMATS = ['NETCDF3_CLASSIC', 'NETCDF4_CLASSIC', 'NETCDF4']
+
+
+# Does this belong elsewhere?
+os.environ["HDF5_USE_FILE_LOCKING"] = "FALSE"
```

Comment:
It would be better to use a context manager or decorator on the test, something along the lines of https://stackoverflow.com/questions/2059482/python-temporarily-modify-the-current-processs-environment

Reply:
good idea. I think we can actually do this with pytest/monkeypatch.